### PR TITLE
I2c slave

### PR DESCRIPTION
**Pull request by kilograham**
* add (slightly modified) pico_i2c_slave library from https://github.com/vmilea/pico_i2c_slave
* Introduce `VTABLE_FIRST_IRQ` constant

### CODE REVIEW

**Comment by kilograham** [comment]
cc @vmilea

File: src/rp2_common/pico_i2c_slave/include/pico/i2c_slave.h [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+
+#include "hardware/i2c.h"
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/** \file pico/i2c_slave.h
+ * \defgrup pico_i2c_slave pico_i2c_slave
+ * \brief I2C slave helper library, which takes care of hooking the I2C IRQ and calling bcak the user with I2C events.
```
Comment (peterharperuk) [inline src/rp2_common/pico_i2c_slave/include/pico/i2c_slave.h:19]: bcak -> back, get rid of todo?

**Review by peterharperuk — DISMISSED** [review]
Minor comment. But looks ok to me

**Comment by peterharperuk** [comment]
https://github.com/raspberrypi/pico-examples/pull/320

**Review by andygpz11 — APPROVED** [review]
Looks good to me and works with the 'i2c slave example' from #320, just as the original did.

**Review by andygpz11 — APPROVED** [review]
I'd missed that but all good now.

**Pull request merged**